protocol/lavaprotocol/finalization_consensus.go

```diff
@@ -100,83 +99,113 @@ func (fc *FinalizationConsensus) insertProviderToConsensus(blockDistanceForFinal
 // create new consensus group if no consensus matched
 // check for discrepancy with old epoch
 // checks if there is a consensus mismatch between hashes provided by different providers
-func (fc *FinalizationConsensus) UpdateFinalizedHashes(blockDistanceForFinalizedData int64, providerAddress string, finalizedBlocks map[int64]string, req *pairingtypes.RelaySession, reply *pairingtypes.RelayReply) (finalizationConflict *conflicttypes.FinalizationConflict, err error) {
+func (fc *FinalizationConsensus) UpdateFinalizedHashes(blockDistanceForFinalizedData int64, consumerAddress sdk.AccAddress, providerAddress string, finalizedBlocks map[int64]string, relaySession *pairingtypes.RelaySession, reply *pairingtypes.RelayReply) (finalizationConflict *conflicttypes.FinalizationConflict, err error) {
+	logSuccessUpdate := func() {
+		if debug {
+			utils.LavaFormatDebug("finalization information update successfully",
+				utils.LogAttr("specId", fc.specId),
+				utils.LogAttr("finalizationData", finalizedBlocks),
+				utils.LogAttr("currentBlockToHashesToAgreeingProviders", fc.currentBlockToHashesToAgreeingProviders),
```

Comment:
this is confusing that its a function before the lock is triggered and only later we lock. can create mistakes add a comment explaining or move it after the lock.

Reply:
Made it more clear in 374cc166891344c2915d97bc58a352ab62ddf921